server/middleware/auth.ts

```diff
@@ -10,9 +10,29 @@ export const checkUser: Middleware = async (req, _res, next) => {
   const settings = getSettings();
   let user: User | undefined | null;
 
-  if (req.header('X-API-Key') === settings.main.apiKey) {
-    const userRepository = getRepository(User);
+  const userRepository = getRepository(User);
+  let trustedProxy = false;
+  // Client IP addresses are appended in this header.
+  // The first header should be the client IP and the last header
+  // should be the address of proxy just upstream of us. We use that
+  // address to figure out if it should be trusted
+  let proxyIP = null;
+
+  const ipHeader = req.header('X-Forwarded-For');
+  if (ipHeader) {
+    const lastComma = ipHeader.lastIndexOf(',');
+    if (lastComma != -1) {
+      proxyIP = ipHeader.slice(lastComma + 1).trim();
+    }
+  }
 
+  if (proxyIP && proxyIP.indexOf('.') != -1) {
```

Comment:
Since we have `ip-address`, we can use `isValid` instead of `indexOf` to perform proper validation.

Reply:
We do not have to do validation here. The only way to reach this code is if the header was set by a trusted proxy and it is assumed to be correct. The only reason we do indexOf is to check if it's a v4 or a v6 address and then search the right store for it.